tests/integration-tests/conftest.py

```diff
@@ -377,8 +377,41 @@ def api_client(region, api_server_factory, api_uri):
         yield api_client_instance
 
 
-def _write_cluster_config_to_outdir(request, cluster_config):
+@pytest.fixture(scope="class")
+def images_factory(request):
+    """
+    Define a fixture to manage the creation and destruction of images.
+
+    The configs used to create clusters are dumped to output_dir/images_configs/{test_name}.config
+    """
+    factory = ImagesFactory()
+
+    def _image_factory(image_id, image_config, region):
+        image_config_file = _write_config_to_outdir(request, image_config, is_cluster=False)
+        image = Image(
+            image_id="-".join([image_id, request.config.getoption("stackname_suffix")])
+            if request.config.getoption("stackname_suffix")
+            else image_id,
+            config_file=image_config_file,
+            region=region,
+        )
+        result = factory.create_image(image)
+        if "BUILD_IN_PROGRESS" not in result:
+            logging.error("image %s creation failed", image_id)
+            image.keep_logs = True
+
+        return image
+
+    yield _image_factory
+    factory.destroy_all_images()
+
+
+def _write_config_to_outdir(request, config, is_cluster=True):
```

Comment:
why using a flag?

You can simply pass a parameter called "config_dir" that will be "clusters_configs" or "image_configs".

With a flag you're adding the knowledge about cluster and image in the write_config_to_outdir method that could instead be a generic method to write a config into a folder.